# CVE-2021-35042
本文件夹内项目提供了`CVE-2021-35042`漏洞的环境.

## 受影响的 Django 版本
 - version 3.1
 - version 3.2

## 使用说明
1. 安装含有`CVE-2021-35042`漏洞的`Django`版本, 测试用的是`3.2.4`.

```shell
python -m pip install Django==3.2.4 -i https://pypi.tuna.tsinghua.edu.cn/simple
```

2. 创建`Mysql`数据库`vuln`.

```shell
CREATE DATABASE vuln;
```

3. 修改`CVE-2021-35042/app.py`中的数据库配置, 将数据库、用户名、密码、端口等都修改为自己的.

```python
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.mysql',
        'NAME': 'vuln',         # 数据库名称
        'USER': 'vuln',     # 数据库用户
        'PASSWORD': '20010728', # 数据库用户密码
        'HOST': '127.0.0.1',    # 数据库地址
        'PORT': '3306',         # 数据库端口
    }
}
```

4. 通过`Django`加载`mysql`数据并运行项目.

```shell
python app.py migrate
python app.py loaddata user.json
python app.py runserver 0.0.0.0:8000
```

## 参考
 - [CVE-2021-35042](https://github.com/vulhub/vulhub/tree/master/django/CVE-2021-35042)